starknet_py/tests/e2e/mock/compile_contracts.sh

```diff
@@ -1,32 +1,70 @@
 #!/bin/bash
 
-MOCK_DIRECTORY=starknet_py/tests/e2e/mock
-CONTRACTS_DIRECTORY="$MOCK_DIRECTORY"/contracts
-CONTRACTS_COMPILED_DIRECTORY="$MOCK_DIRECTORY"/contracts_compiled
+compile_contracts_with_scarb() {
+    if ! command -v asdf >/dev/null 2>&1; then
+        echo "Install asdf, before executing this script is required to manage scarb version"
+        exit 1
+    fi
 
-# delete all artifacts except precompiled ones
-find $CONTRACTS_COMPILED_DIRECTORY -maxdepth 1 -type f -delete
+    asdf plugin-add scarb
+    asdf install scarb 0.4.1
+    asdf install scarb 2.6.3
 
-# compile Cairo test contracts
-echo "Compiling Cairo contracts with $(poetry run starknet-compile-deprecated --version)"
+    output=$(find . -type f -name "Scarb.toml" -execdir sh -c '
+        echo "Running \"scarb fmt\" in directory: $PWD"
+        find target/dev -maxdepth 1 -type f -delete
+        scarb build
+    ' \;)
```

Comment:
this is overcomplicated - it is suboptimal to run find inside a find. I think scarb recompiles at each run, so we probably should not bother with artifacts removal